PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,880 @@
+        """Get or create masking view from component elements.
+
+        :param device_ids: list of device ids
+        :param source_sg: the source sg
+        :param target_sg: the target sg
+        :param create_vol_f: create volume flag
+        :returns: string
+        """
+        print_str = '\nMoving %d volume(s) from %s to %s.'
+        self.smart_print(
+            print_str, DEBUG, len(device_ids), source_sg, target_sg)
+        self.smart_print(
+            '\nPlease be patient, this may take several minutes...',
+            DEBUG)
+        # Create a small volume
+        if create_vol_f:
+            self.conn.provisioning.create_volume_from_sg_return_dev_id(
+                'last_vol', source_sg, '1')
+        # Move the volume from the old storage group to the
+        # new storage group
+        message = self.conn.provisioning.move_volumes_between_storage_groups(
+            device_ids, source_sg, target_sg,
+            force=False
+        )
+        return message
+
+    def validate_list(self, full_list, sub_list):
+        """Validate the sub list is within the full list.
+
+        :param full_list: full list
+        :param sub_list: sub list
+        :returns: Boolean
+        """
+        return True if all(elem in full_list for elem in sub_list) else False
+
+    def choose_subset_volumes(self, storagegroup, volume_list):
+        """Validate the sub list is within the full list.
+
+        :param storagegroup: full list
+        :param volume_list: sub list
+        :returns: volume_list
+        """
+        create_vol = False
+        self.smart_print('Here is the full list of volumes in SG %s: %s',
+                         DEBUG, storagegroup, volume_list)
+        txt = ('Which do you want to migrate (comma separated list): ',
+               volume_list)
+        txt_out = self.input(txt)
+        if txt_out:
+            sub_volume_list = txt_out.split(',')
+            sub_volume_list = [x.strip(' \'') for x in sub_volume_list]
+            if self.validate_list(volume_list, sub_volume_list):
+                if len(volume_list) == len(sub_volume_list):
+                    create_vol = True
+                volume_list = sub_volume_list
+            else:
+                print_str = ('Unable to validate your list, '
+                             'no volumes will be migrated.')
+                self.smart_print(print_str, WARNING)
+                volume_list = []
+        else:
+            self.smart_print('You cannot input an empty list', DEBUG)
+            txt = 'Do you want to choose again Y/N or X:'
+            txt_out = self.input(txt)
+            if self.check_input(txt_out, 'Y'):
+                volume_list, create_vol = self.choose_subset_volumes(
+                    storagegroup, volume_list)
+            else:
+                sys.exit()
+        return volume_list, create_vol
+
+    def get_volume_list(self, storagegroup):
```

Comment:
`storage_group_name `and `storagegroup` are used interchangeably throughout, might be better for documentation if they are the same but if we don't expect people to dig into these files and use methods but instead just the migrate script as a whole then it doesn't matter